phpunit/class-block-library-navigation-fallbacks-test.php

```diff
@@ -0,0 +1,370 @@
+		$this->assertEquals( 'Navigation', $fallback->post_title, 'The fallback Navigation Menu should be have the expected title.' );
+		$this->assertEquals( '<!-- wp:page-list /-->', $fallback->post_content, 'The fallback Navigation Menu should contain a Page List block.' );
+		$this->assertEquals( 'publish', $fallback->post_status, 'The fallback Navigation Menu should be published.' );
+
+		$navs_in_db = $this->get_navigations_in_database();
+		$this->assertCount( 1, $navs_in_db, 'A single Navigation Menu should exist.' );
+	}
+
+	/**
+	 * @covers ::block_core_navigation_create_fallback
+	 */
+	public function test_creates_fallback_from_existing_navigation_menu_even_if_classic_menu_exists() {
+
+		// Create a Navigation Post.
+		$navigation_post = self::factory()->post->create_and_get(
+			array(
+				'post_type'    => 'wp_navigation',
+				'post_title'   => 'Existing Navigation Menu',
+				'post_content' => '<!-- wp:page-list /-->',
+			)
+		);
+
+		// Also create a Classic Menu - this should be ignored.
+		$menu_id = wp_create_nav_menu( 'Existing Classic Menu' );
+
+		$fallback = gutenberg_block_core_navigation_create_fallback();
+
+		$this->assertEquals( $fallback->post_title, $navigation_post->post_title, 'The title of the fallback Navigation Menu should match that of the existing Navigation Menu.' );
+		$this->assertEquals( $fallback->post_type, $navigation_post->post_type, 'The post type of the fallback Navigation Menu should match that of the existing Navigation Menu.' );
+		$this->assertEquals( $fallback->post_content, $navigation_post->post_content, 'The contents of the fallback Navigation Menu should match that of the existing Navigation Menu.' );
+		$this->assertEquals( $fallback->post_status, $navigation_post->post_status, 'The status of the fallback Navigation Menu should match that of the existing Navigation Menu.' );
+
+		$navs_in_db = $this->get_navigations_in_database();
+		$this->assertCount( 1, $navs_in_db, 'A single Navigation Menu should exist.' );
+
+		// Cleanup.
+		wp_delete_nav_menu( $menu_id );
+	}
+
+	/**
+	 * @covers ::block_core_navigation_create_fallback
+	 */
+	public function test_should_skip_if_filter_returns_truthy() {
+		add_filter( 'block_core_navigation_skip_fallback', '__return_true' );
+
+		gutenberg_block_core_navigation_create_fallback();
+
+		$navs_in_db = $this->get_navigations_in_database();
+		$this->assertCount( 0, $navs_in_db, 'No Navigation Menus should have been created.' );
+
+		remove_filter( 'block_core_navigation_skip_fallback', '__return_true' );
+	}
+
+
+	/**
+	 * @covers ::gutenberg_block_core_navigation_get_fallback_blocks
+	 */
+	public function test_should_get_fallback_blocks_when_no_navigations_exist() {
+		$fallback_blocks = gutenberg_block_core_navigation_get_fallback_blocks();
+
+		$this->assertIsArray( $fallback_blocks, 'Fallback blocks should be an array.' );
+		$this->assertEmpty( $fallback_blocks, 'Fallback blocks should be empty.' );
```

Comment:
It would be better to replace these two assertions with a single one, although it's just my personal preference.
```suggestion
		$this->assertSame( array(), $fallback_blocks, 'Fallback blocks should be an empty array.' );
```